onnxruntime/test/providers/cpu/math/element_wise_ops_test.cc

```diff
@@ -11,6 +11,41 @@
 namespace onnxruntime {
 namespace test {
 
+TEST(BroadcastingTest, DimWithZeroHandling) {
+  auto run = [](OpTester& tester) {
+    // exclude NGraph and TensorRT as this isn't handled by those EPs
+    tester.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider, kNGraphExecutionProvider});
+  };
+
+  // equal rank
+  OpTester test("Add");
```

Comment:
should we have a case with a scalar (shape: ()) and an input with a zero dim value ?

Reply:
added